api/envoy/extensions/filters/http/rate_limit_quota/v3/rate_limit_quota.proto

```diff
@@ -0,0 +1,416 @@
+  // If not set, inherits the value from the less specific definition.
+  //
+  // See usage example: :ref:`RateLimitQuotaFilterConfig.bucket_matchers
+  // <envoy_v3_api_field_extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaFilterConfig.bucket_matchers>`.
+  xds.type.matcher.v3.Matcher bucket_matchers = 2;
+}
+
+// Rate Limit Quota Bucket Settings to apply on the successful ``bucket_matchers`` match.
+//
+// Specify this message in the :ref:`Matcher.OnMatch.action
+// <envoy_v3_api_field_.xds.type.matcher.v3.Matcher.OnMatch.action>` field of the
+// ``bucket_matchers`` matcher tree to assign the matched requests to the Quota Bucket.
+// Usage example: :ref:`RateLimitQuotaFilterConfig.bucket_matchers
+// <envoy_v3_api_field_extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaFilterConfig.bucket_matchers>`.
+// [#next-free-field: 6]
+message RateLimitQuotaBucketSettings {
+  // Configures the behavior after the first request has been matched to the bucket, and before the
+  // the RLQS server returns the first quota assignment.
+  message NoAssignmentBehavior {
+    // Blocking behavior: wait for the RLQS server to return the first assignment before
+    // processing any requests.
+    // [#not-implemented-hide:]
+    message BlockingBehavior {
+      // Limit the time the requests are blocked until the RLQS server to returns
+      // the first assignment. If the server doesn't respond within the configured time,
+      // all pending requests will be processed according to the :ref:`on_timeout
+      // <envoy_v3_api_field_extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaBucketSettings.NoAssignmentBehavior.BlockingBehavior.on_timeout>`
+      // field.
+      google.protobuf.Duration timeout = 1 [(validate.rules).duration = {
+        required: true
+        gt {}
+      }];
+
+      // Allow or Deny the requests after the duration specified in the :ref:`timeout
+      // <envoy_v3_api_field_extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaBucketSettings.NoAssignmentBehavior.BlockingBehavior.timeout>`
+      // field reached.
+      //
+      // If unset, the default behavior is to allow all requests (:ref:`BlanketRule.ALLOW_ALL
+      // <envoy_v3_api_enum_value_type.v3.RateLimitStrategy.BlanketRule.ALLOW_ALL>`).
+      type.v3.RateLimitStrategy.BlanketRule on_timeout = 2;
+    }
+
+    oneof no_assignment_behavior {
+      option (validate.required) = true;
+
+      // Apply the rate limiting strategy to the requests matched into the bucket
+      // until the RLQS server returns the first assignment.
+      type.v3.RateLimitStrategy fallback_rate_limit = 1;
+
+      // Block all requests matched into the bucket until the RLQS server returns
+      // the first quota assignment.
+      // [#not-implemented-hide:]
+      BlockingBehavior block_requests = 2;
+    }
+  }
+
+  // Specifies the behavior when the bucket's assignment has expired, and can't be refreshed
+  // due to the RLQS server's unavailability.
```

Comment:
It's not just about unavailability right? Given the design the RLQS server may just not send anything? Can you clarify?

Reply:
Can you clarify in the text also?

Reply:
Clarified in https://github.com/envoyproxy/envoy/pull/19793/commits/f7665e145b75fb886c124ed6beaa1df35cd9fd8a.